src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -394,6 +395,10 @@ public function load(array $configs, ContainerBuilder $container)
             // remove tagged iterator argument for resource checkers
             $container->getDefinition('config_cache_factory')->setArguments([]);
         }
+
+        if ($config['disallow_search_engine_index'] ?? false) {
+            $container->register(DisallowRobotsIndexingListener::class)->addTag('kernel.event_subscriber');
```

Comment:
Should be move to `debug.xml` which is loaded when debug mode is enabled (which is exactly the behavior we are looking for). This piece of code should be about removing the tag.

Reply:
The thing is you might want to enable this feature, but without enabling `debug`: pre-prod accessible from the outside, costumer portal, ... Should I put it there anyway?

Reply:
Understood :) So, the registration of the service should be moved to an XML file and the tag removed when the config disables it.

Reply:
Moved to `web.xml` then